### PR TITLE
EKS fix validation when no credential is found, cluster name display

### LINKED ISSUE #11199: EKS cluster edit is not working
**Setup**
- Rancher version: v2.9-16190ecee0aa3fb4aa570aa0fc932a36d9dcd082-head
- Browser type & version: Version 123.0.6312.107 (Official Build) (arm64)

**Describe the bug**
Once you try to edit an existing EKS cluster, the save button in the UI is locked for all the settings, until you change the cluster name. Once the user changes the cluster name and hit save, the UI displays an error message: `{"data":""}` and an internal server error 500 is shown in the console

**To Reproduce**
1- From the rancher dashboard, go to Clusters Management > Create > Amazon EKS
2- Create an EKS cluster 
3- While the cluster is provisioning, try to edit the cluster config via the UI
4- Change the values of all the fields available for edition, except the cluster name
5- Then, change the cluster name and hit the Save button
6- Observe the error output displayed in the UI

**Result**
The Save button is just enabled in the UI if the user changes the cluster name. For changes in any of the other fields, the Save button remains locked. Once, the user changes the cluster name, and hit save, an error occurs.

**Expected Result**
The save button should be available for every single change made in the cluster config and any changes must be successfully saved.

**Screenshots**

https://github.com/rancher/dashboard/assets/70187881/7fa744e0-c94a-498a-bf62-05aa46d15759

Screenshot 2024-06-07 at 7 41 10 PM

### PR DESCRIPTION
**Pull request by mantis-toboggan-md**
### Summary
Fixes #11199 (see comment for details)
#11105 

### Occurred changes and/or fixed issues
This PR updates EKS form validators to not run until the form has renders. It also updates which name property is displayed in the name input, see https://github.com/rancher/dashboard/issues/11199#issuecomment-2163774098. This PR also has some tweaks to tab name/labels so there are no console errors thrown if the user tries to use duplicate node group names.

### Technical notes summary
Validators have been moved out of CruEKS to  write unit tests without having to test form validation mixin functionality.

### Areas or cases that should be tested
1. Create a cluster and verify that the POST request has correctly set both `cluster.name` and `cluster.eksConfig.displayName`
2. Edit the cluster and update the name; verify that both `cluster.name` and `cluster.eksConfig.displayName` are updated

Node groups:
1. Add a node pool and give it the same name as the first pool: the save button should be greyed out and an input-level error displayed. No console errors from tabbed.
2. Clear a node group's name: save button should be disabled and an input-level error shown. No console errors from tabbed. The tab should be labeled 'unnamed node group'

### Checklist
- [ x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: pkg/eks/components/CruEKS.vue [hunk 1]
```diff
@@ -253,91 +255,22 @@ export default defineComponent({
   computed: {
     ...mapGetters({ t: 'i18n/t' }),
 
+    needsValidation() {
```
Comment (momesgin) [inline pkg/eks/components/CruEKS.vue:258]: if I understand correctly this is used inside each validator as a guard, but wouldn't  inside the `fvExtraRules` be a better place right before returning the validation object?

We also have `hasCredential` as a computed property that returns the exact same value, can it be re-used instead?

File: pkg/eks/util/validators.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
+import { EKSConfig, EKSNodeGroup } from 'types';
```
Comment (momesgin) [inline pkg/eks/util/validators.ts:1]: I really like moving them here!

File: pkg/eks/components/CruEKS.vue (hunk 1 above)
Comment (mantis-toboggan-md) [reply pkg/eks/components/CruEKS.vue]: These are both great suggestions. I've updated the PR accordingly.

File: pkg/eks/util/__tests__/validators.test.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
+import EKSValidators, { CruEKSContext } from '../validators';
+
+const mockTranslation = (key: string) => key;
+
+describe('validate EKS Cluster name', () => {
+  it('should return an error if displayName is an empty string or undefined', () => {
+    const ctx = {
+      config:          { displayName: '' },
+      t:               mockTranslation,
+      needsValidation: true
```
Comment (momesgin) [inline pkg/eks/util/__tests__/validators.test.ts:10]: `needsValidation ` can be removed from these tests too

**Review by momesgin — APPROVED** [review]
lgtm

**Pull request merged**